src/Features/VisualBasic/Portable/DocumentSymbols/VisualBasicDocumentSymbolsService.vb

```diff
@@ -0,0 +1,223 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.DocumentSymbols
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.PooledObjects
+Imports Microsoft.CodeAnalysis.Tags
+Imports Microsoft.CodeAnalysis.Text
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.DocumentSymbols
+
+    <ExportLanguageService(GetType(IDocumentSymbolsService), LanguageNames.VisualBasic), [Shared]>
+    Friend Class VisualBasicDocumentSymbolsService
+        Inherits AbstractDocumentSymbolsService
+
+        Private Shared ReadOnly s_typeFormat As SymbolDisplayFormat = New SymbolDisplayFormat(
+            genericsOptions:=SymbolDisplayGenericsOptions.IncludeTypeParameters Or SymbolDisplayGenericsOptions.IncludeVariance)
+
+        Private Shared ReadOnly s_memberFormat As SymbolDisplayFormat = New SymbolDisplayFormat(
+            genericsOptions:=SymbolDisplayGenericsOptions.IncludeTypeParameters,
+            memberOptions:=SymbolDisplayMemberOptions.IncludeParameters Or SymbolDisplayMemberOptions.IncludeType,
+            parameterOptions:=SymbolDisplayParameterOptions.IncludeType,
+            miscellaneousOptions:=SymbolDisplayMiscellaneousOptions.UseSpecialTypes)
+
+        Private Shared ReadOnly s_constantTag As ImmutableArray(Of String) = ImmutableArray.Create(WellKnownTags.Constant)
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+        End Sub
+
+        Protected Overrides Function GetSymbol(semanticModel As SemanticModel, node As SyntaxNode, cancellationToken As CancellationToken) As ISymbol
```

Comment:
Ditto as C#: why not just call GetDeclaredSymbol() and just pass in the base node?

Reply:
(this seems like a guaranteed bug every time the compiler invents something new...)